### ISSUE TITLE
'cf api' and 'target' displays cached api version instead of current remote CC API

### ISSUE THREAD

**Issue by gberche-orange** [cloudfoundry/cli#558]
It may be confusing to see that `cf api` does not display the current api version for the current api endpoint, but instead a cached version of this stored into $CF_HOME/.cf/config.json (key ApiVersion). This is only refreshed when resetting the api.

The command displays a cached outdated api version

```
$ cf api
API endpoint: https://api.run.pivotal.io (API version: 2.25.0)
```

after resetting the api endpoint, the command now displays the correct api version

```
$ cf api https://api.run.pivotal.io
Setting api endpoint to https://api.run.pivotal.io...
OK

API endpoint:   https://api.run.pivotal.io (API version: 2.34.0)
User:           bercheg@gmail.com
No org or space targeted, use cf.exe target -o ORG -s SPACE
```

It might make sense to either:
* systematically recall v2/info when cf api is called and display a fresh version returned by the v2/info endpoint (and not displaying the api version in `cf target`)
* only display the current api version and additional meta-data into a distinct new command (e.g. `cf info`) targetting v2/info endpoint

**Comment by camelpunch** [comment]
Thanks for the report. Since it's now in Tracker, you can track its progress there.

**Comment by dkoper** [comment]
Currently `cf target` and `cf api` without arguments display the most recently targeted org, space and api (and version), and not making any API calls. And they could get stale as you pointed out. In fact, this api version number is used by various commands or flags to check if the backend supports the feature and fail if too low.

I think in most cases/for most people, this information doesn't go stale that quickly, so I'm not sure we'd want to slow down this command, or every other command, with API calls to keep these fields up to date?

Would a tip on the help or output not be sufficient to take away any confusion?
```
D:\Users\koperd\Downloads\bin>cf target
Cached details of your target:
API endpoint:   https://api.mycf.com (API version: 2.47.0)
User:           diesk
Org:            dies-test
Space:          dev
```

or

```
D:\Users\koperd\Downloads\bin>cf h api
NAME:
   api - Set or view cached target api url
...
```

**Comment by gberche-orange** [comment]
@dkoper Sorry for late reply on this. Your suggested tip on the help and command output would certainly help remove the confusion on the ```cf api``` command. 

I would suggest to complete the tip to indicate how to refresh the cached api version, especially when the CLI rejects a command because of API version incompatibility: 

I recently got a message from a command asking for the latest api version. Trying to logout and login did not refresh the version. "cf api" with the same target did refresh the cached version, but felt unintuitive.

Suggesting something like ```API version X.Y required. Try to refresh the cached api meta-data with cf api $target, or request a CC API update to the CF Platform team.```

Even best, on an incompatible API error, the CLI could systematically attempt to refresh the cached api version to the expense of a potentially extra wasted ```/v2/info``` call, as to avoid manual refresh by the end-user.

**Comment by dkoper** [comment]
Hi @gberche-orange 

I have created a separate story for your last suggestion (refresh on incompatible API error) - thanks for the suggestion!
https://www.pivotaltracker.com/story/show/115650433

Leaving this issue and corresponding story open for the issue that the displayed API version may be stale.

**Comment by gberche-orange** [comment]
great, thanks @dkoper

**Comment by dkoper** [comment]
Hi @gberche-orange,

The just released cf CLI v6.17.0 implements your last suggestion (refreshing CC version before failing):
https://github.com/cloudfoundry/cli#downloads

Thanks again for the suggestion!
Dies Koper
CF CLI PM

**Comment by gberche-orange** [comment]
Thanks a lot @dkoper ! 

Tested it successfully with the "cf unbind-route-service" command.
